app/notification.go

```diff
@@ -1338,28 +1338,31 @@ type CRTNotifiers struct {
 	Push model.StringArray
 }
 
-func (c *CRTNotifiers) addUserToNotify(user *model.User, mentions *ExplicitMentions) {
+func (c *CRTNotifiers) addFollowerToNotify(user *model.User, mentions *ExplicitMentions, channelMemberNotificationProps model.StringMap, channel *model.Channel) {
 	// user notify props
-	desktop := user.NotifyProps[model.DesktopNotifyProp]
-	push := user.NotifyProps[model.PushNotifyProp]
-	shouldEmail := user.NotifyProps[model.EmailNotifyProp] == "true"
+	_, userWasMentioned := mentions.Mentions[user.Id]
 
-	// user thread notify props
-	desktopThreads := user.NotifyProps[model.DesktopThreadsNotifyProp]
-	emailThreads := user.NotifyProps[model.EmailThreadsNotifyProp]
-	pushThreads := user.NotifyProps[model.PushThreadsNotifyProp]
+	notifyDesktop, notifyPush, notifyEmail := user.ShouldNotifyCRT(userWasMentioned)
 
-	_, userWasMentioned := mentions.Mentions[user.Id]
+	notifyChannelDesktop, notifyChannelPush := model.ShouldChannelMemberNotifyCRT(channelMemberNotificationProps, userWasMentioned)
 
-	if desktop != model.UserNotifyNone && (userWasMentioned || desktopThreads == model.UserNotifyAll || desktop == model.UserNotifyAll) {
+	if channelMemberNotificationProps[model.DesktopNotifyProp] == model.ChannelNotifyDefault {
```

Comment:
I think the 3 slice populations can be simplified further. Firstly, the action in both conditions are same, so they can be shortened to:

```go
if (channelMemberNotificationProps[model.DesktopNotifyProp] == model.ChannelNotifyDefault && notifyDesktop) || notifyChannelDesktop {
  c.Desktop = append(c.Desktop, user.Id)
}
```

Secondly, this entire logic can be moved inside `shouldChannelMemberNotifyCRT`. This method can just append `c.Desktop`, `c.Email` and `c.Push`.

Also, let's write some comments to explain the code in `ShouldChannelMemberNotifyCRT` once the code is moved inside that. I am finding it hard to understand so many if conditions.

Reply:
> I think the 3 slice populations can be simplified further. Firstly, the action in both conditions are same, so they can be shortened to:

It's not the same, the rule is if the channel notify prop is set to default respect **only** the user global setting,
in any other case respect the channel member setting.

Reply:
> Secondly, this entire logic can be moved inside `shouldChannelMemberNotifyCRT`. This method can just append `c.Desktop`, `c.Email` and `c.Push`.

Since the logic is somewhat complicated having 2 different checks for notify props, and one that selects between the both was easier for me to reason about.

Reply:
> It's not the same, the rule is if the channel notify prop is set to default respect only the user global setting,
in any other case respect the channel member setting.

Ah my bad. Then let's just merge the first 2 conditions.

Reply:
> Ah my bad. Then let's just merge the first 2 conditions.

you mean these 2?

```go
channelMemberNotificationProps[model.DesktopNotifyProp] == model.ChannelNotifyDefault && notifyDesktop
```

Reply:
Yep.

Reply:
done

Reply:
I see that you actually merged all the conditions. Did something change?